clang_tidy/clang_tidy.bzl

```diff
@@ -151,7 +125,47 @@ def _toolchain_flags(ctx, action_name = ACTION_NAMES.cpp_compile):
     )
     return flags
 
-def _safe_flags(flags):
+def deps_flags(ctx, deps, *, escape):
+    compilation_contexts = [dep[CcInfo].compilation_context for dep in deps]
+    additional_files = depset(transitive = [
+        compilation_context.headers
+        for compilation_context in compilation_contexts
+    ])
+
+    flags = []
+    for compilation_context in compilation_contexts:
+        # add defines
+        for define in compilation_context.defines.to_list():
+            if escape:
+                flags.append("-D'" + define + "'")
+            else:
```

Comment:
this is pretty annoying, in our project we have some defines like `-DFOO=<header.h>` which need escaping. Args objects do that automatically but the shell does not. We can't use args objects in the test because they are expected to be used for ctx.actions.run and don't end up in the runfiles